### PR TITLE
feat(1314): Sd local

### PR DESCRIPTION
**Pull request by s-yoshika**
## Context
Add guides for sd-local.

## Objective
This add the guide for sd-local.

Note:
- In `Install sd-local` section, we describe how to install with `go get`, but we'll change it if sd-local is released with https://github.com/screwdriver-cd/sd-local/pull/17

## License

I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: docs/user-guide/local.md [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+  url: "#simple-usage"
+- title: config command
+  url: "#config-command"
+- title: build command
+  url: "#build-command"
+---
+
+# What is sd-local ?
+sd-local enable you to build as Screwdriver does without uploading source code to SCM.
+You can use features including Command, Template, Metadata and so on as same as Screwdriver.
```
Comment (tkyi) [inline docs/user-guide/local.md]:
```suggestion
sd-local enables running a more feature-complete local Screwdriver without needing uploading source code to SCM. With sd-local, you can use features such as Command, Template, Metadata, and more.
```

File: docs/user-guide/local.md [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+- [docker](https://www.docker.com/)
+  - Use Docker container as build environment.
+- [git](https://git-scm.com/)
+  - Required when using [--src-url] option.
+
+
+## How to Install
+
+
+You can install sd-local by the following command.
```
Comment (tkyi) [inline docs/user-guide/local.md]:
```suggestion
You can install sd-local using the following command:
```

File: docs/user-guide/local.md [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+```bash
+$ go get github.com/screwdriver-cd/sd-local
+```
+
+
+Note:
+go command is required to install sd-local.
+https://golang.org/
```
Comment (tkyi) [inline docs/user-guide/local.md]: Can this section be combined with other Requirements?

File: docs/user-guide/local.md [hunk 4]
```diff
@@ -0,0 +1,213 @@
...
+
+
+# Simple usage
+This section describes the steps to execute build with sd-local.
+Let Screwdriver API be `https://api.screwdriver.cd`, and let Screwdriver Store be `https://store.screwdriver.cd` in this section.
```
Comment (tkyi) [inline docs/user-guide/local.md]:
```suggestion
This section describes the steps needed to execute a build with sd-local.
Let Screwdriver API be `https://api.screwdriver.cd`, and let Screwdriver Store be `https://store.screwdriver.cd` in this section.
```

File: docs/user-guide/local.md [hunk 5]
```diff
@@ -0,0 +1,213 @@
...
+## Get repository for build
+You have to get source code and `screwdriver.yaml` for build. You can use [quickstart-generic](https://github.com/screwdriver-cd-test/quickstart-generic.git).
+```bash
+$ git clone https://github.com/screwdriver-cd-test/quickstart-generic.git
+$ cd quickstart-generic
+```
+
+
+## Build configuration
+sd-local needs some settings for builds, so you have to configure with config command.
```
Comment (tkyi) [inline docs/user-guide/local.md]:
```suggestion
sd-local needs some settings for builds, so you have to configure using the `config` command:
```

File: docs/user-guide/local.md [hunk 6]
```diff
@@ -0,0 +1,213 @@
...
+```bash
+$ sd-local config set api-url https://api.screwdriver.cd
+$ sd-local config set store-url https://store.screwdriver.cd
+$ sd-local config set token <API Token>
+$ sd-local config set launcher-version latest
+```
+
+
+## Execute build
+You can execute the main job using sd-local.
```
Comment (tkyi) [inline docs/user-guide/local.md]:
```suggestion
You can execute the main job using sd-local `build` command:
```

**Comment by s-yoshika** [comment]
@tkyi Thank you for your review and fix them.

And also we fix the guid of installing because now sd-local can be installed without go tool.

File: docs/ja/user-guide/local.md [hunk 7]
```diff
@@ -0,0 +1,184 @@
...
+# sd-localとは？
+sd-localではSCMにコードをアップロードせずに、手元でScrewdriverと同じビルドを再現することができます。sd-localは、Command, Template, Metadataといった機能をScrewdriverと同様に利用することができます。
+
+# sd-localのインストール
+## 実行環境について
+sd-local実行には以下のコマンドが必要になります
+- [docker](https://www.docker.com/)
+  - ビルドの実行環境としてdockerコンテナを利用します
+- [git](https://git-scm.com/)
+  - [src-url]オプションを利用する場合に必要になります
```
Comment (tk3fftk) [inline docs/ja/user-guide/local.md]: Is this `[src-url]` intended to a link? Or should we use the format like `--src-url` ?

File: docs/user-guide/local.md [hunk 8]
```diff
@@ -0,0 +1,207 @@
...
+# What is sd-local ?
+sd-local enables running a more feature-complete local Screwdriver without needing uploading source code to SCM. With sd-local, you can use features such as Command, Template, Metadata, and more.
+
+
+# Installing sd-local
+## Requirements
+- [docker](https://www.docker.com/)
+  - Use Docker container as build environment.
+- [git](https://git-scm.com/)
+  - Required when using [--src-url] option.
```
Comment (tk3fftk) [inline docs/user-guide/local.md]: Is this `[src-url]` intended to a link?  Or should we using backquotes?

File: docs/user-guide/local.md [hunk 9]
```diff
@@ -0,0 +1,207 @@
...
+You can configure sd-local settings in key/value format by config command.
+Your settings are stored in `~/.sdlocal/config`.
+
+
+## Usage
+
+
+### set subcommand
+You can configure sd-local in key/value format. You have to set `api-url`, `store-url` and `token` to execute builds.
+Please refer to [List of the keys](#list-of-the-key) about available settings.
```
Comment (tkyi) [inline docs/user-guide/local.md]:
```suggestion
Please refer to the [List of keys](#list-of-keys) about available settings.
```

File: docs/user-guide/local.md [hunk 10]
```diff
@@ -0,0 +1,207 @@
...
+### view subcommand
+You can confirm the current settings.
+
+
+```bash
+$ sd-local config view
+```
+
+
+### List of the key.
```
Comment (tkyi) [inline docs/user-guide/local.md]:
```suggestion
### List of keys
You must set `api-url`, `store-url`, and `token` in order to execute builds.
```

File: docs/user-guide/local.md [hunk 11]
```diff
@@ -0,0 +1,207 @@
...
+# config command
+You can configure sd-local settings in key/value format by config command.
+Your settings are stored in `~/.sdlocal/config`.
+
+
+## Usage
+
+
+### set subcommand
+You can configure sd-local in key/value format. You have to set `api-url`, `store-url` and `token` to execute builds.
```
Comment (tkyi) [inline docs/user-guide/local.md]:
```suggestion
You can configure sd-local in key/value format. You must set `api-url`, `store-url`, and `token` in order to execute builds.
```

File: docs/user-guide/local.md [hunk 12]
```diff
@@ -0,0 +1,207 @@
...
+
+### List of the key.
+
+|key|description|
+|---|---|
+|api-url|Please specify the URL of Screwdriver API of the cluster you are using.|
+|store-url|Please specify the URL of Screwdriver Store of the cluster you are using.|
+|token|Please specify the [API token](api#authentication-and-authorization)|
+|launcher-image|Please specify the launcher image. (default: `screwdrivercd/launcher`)|
+|launcher-version|Please specify the launcher version. (default: `stable`)|
```
Comment (tkyi) [inline docs/user-guide/local.md]:
```suggestion
|api-url|The Screwdriver API URL of the cluster you are using|
|store-url|The Screwdriver Store URL of the cluster you are using|
|token|An [API token](api#authentication-and-authorization) for the Screwdriver API of the cluster you are using|
|launcher-image|The launcher image (default: `screwdrivercd/launcher`)|
|launcher-version|The launcher version (default: `stable`)|
```

File: docs/user-guide/local.md [hunk 13]
```diff
@@ -0,0 +1,207 @@
...
+
+
+# build command
+This command run the build in your local environment.
```
Comment (tkyi) [inline docs/user-guide/local.md]:
```suggestion
This command runs builds in your local environment.
```

File: docs/user-guide/local.md [hunk 14]
```diff
@@ -0,0 +1,207 @@
...
+
+
+## Usage
+Run build with specific the job. Even if a job is succeeded, any other jobs are not triggered.
```
Comment (tkyi) [inline docs/user-guide/local.md]:
```suggestion
Run a build for a specific job. Even if the build succeeds, other builds are not triggered.
```

File: docs/user-guide/local.md [hunk 15]
```diff
@@ -0,0 +1,207 @@
...
+```bash
+$ sd-local build <job name>
+```
+
+
+It starts the job based on `screwdriver.yaml` in the current directory, and the build artifacts are created under the "./sd-artifacts" directory.
```
Comment (tkyi) [inline docs/user-guide/local.md]:
```suggestion
It starts the build based on the `screwdriver.yaml` config in the current directory. Build artifacts will be created under the "./sd-artifacts" directory.
```

File: docs/user-guide/local.md [hunk 16]
```diff
@@ -0,0 +1,207 @@
...
+
+
+## Options
+The following options can be used in build command.
```
Comment (tkyi) [inline docs/user-guide/local.md]:
```suggestion
The following options can be used with the build command:
```

File: docs/user-guide/local.md [hunk 17]
```diff
@@ -0,0 +1,207 @@
...
+|--env-file|Set the environment variables in the build environment in file format. (Refer to [env-file format](#env-file-format))|
+|-m, --memory|Specify the limit value of the memory resource of the build environment. (Can be specified in b, k, m, g memory unit)|
+|--meta|Specify [metadata](metadata) used in build environment. (JSON format)|
+|--meta-file|Specify [metadata](metadata) used in build environment in the file. (JSON format)|
+|--src-url|Specify the remote SCM URL to build. (`https` or `ssh` format)|
+
+
+Note:
+- sd-local uses the value specified by `--env` option, if a same environment variable is specified in both `--env` and `--env-file` options.
+- You can't use both `--meta` and `--meta-file` options.
```
Comment (tkyi) [inline docs/user-guide/local.md]:
```suggestion
|option|description|
|---|---|
|--artifacts-dir|The build artifacts destination (default: `./sd-artifacts`)|
|-e, --env|The environment variables in a build environment using `<key>=<value>` format (multiple variables allowed)|
|--env-file|The environment variables in a build environment using file format (refer to [env-file format](#env-file-format))|
|-m, --memory|The memory limit of the build environment (can be specified in b, k, m, or g memory unit)|
|--meta|The [metadata](metadata) used in a build environment using string JSON format|
|--meta-file|The [metadata](metadata) used in build environment using file format|
|--src-url|The remote SCM URL to build (`https` or `ssh` format)|


Note:
- sd-local uses the value specified by `--env` option, if a same environment variable is specified in both `--env` and `--env-file` options.
- You can't use both `--meta` and `--meta-file` options.
```

File: docs/user-guide/local.md [hunk 18]
```diff
@@ -0,0 +1,207 @@
...
+
+
+### env-file format
+`--env-file` option is using the file format as same as the [.env](https://docs.docker.com/compose/env-file/) format of docker.
+
+
+- Each line in an `env` file to be in `VAR=VAL` format.
+- Lines beginning with `#` are processed as comments and ignored.
+- Blank lines are ignored.
+- There is no special handling of quotation marks. This means that they are part of the VAL.
```
Comment (tkyi) [inline docs/user-guide/local.md]:
```suggestion
`--env-file` option expects a file with a file format like the [.env](https://docs.docker.com/compose/env-file/) format of docker.


- Each line in an `env` file should follow `VAR=VAL` format.
- Lines beginning with `#` are processed as comments and ignored.
- Blank lines are ignored.
- There is no special handling of quotation marks. This means that they are part of the VAL.
```

File: docs/user-guide/local.md [hunk 19]
```diff
@@ -0,0 +1,207 @@
...
+
+
+## List of environment variables
+You can use the following environment variables in sd-local build. The default values whose value is showed as  '-' are same as Screwdriver. Please refer to [environment variables](environment-variables) in the details.
```
Comment (tkyi) [inline docs/user-guide/local.md]:
```suggestion
You can use the following environment variables in sd-local build. The default values shown as  `-` are the same defaults as in production Screwdriver. Please refer to [environment variables](environment-variables) for more details.
```

File: docs/user-guide/local.md (hunk 17 above)
Comment (tkyi) [reply docs/user-guide/local.md]: Might be good to have some examples for these, but not necessary

**Comment by tkyi** [comment]
A lot of small grammar suggestions, but otherwise looks good to me 👍

File: docs/user-guide/local.md (hunk 3 above)
Comment (sugarnaoming) [reply docs/user-guide/local.md]: I fixed.

File: docs/ja/user-guide/local.md (hunk 7 above)
Comment (sugarnaoming) [reply docs/ja/user-guide/local.md]: I fixed.

File: docs/user-guide/local.md (hunk 8 above)
Comment (sugarnaoming) [reply docs/user-guide/local.md]: I fixed.

File: docs/user-guide/local.md (hunk 17 above)
Comment (sugarnaoming) [reply docs/user-guide/local.md]: I fixed.
Added example of `--meta`  option.

File: docs/ja/user-guide/local.md [hunk 20]
```diff
@@ -0,0 +1,184 @@
...
+
+# config コマンド
+configコマンドでは、sd-local自体の設定をkey/valueの形式で行います。
+設定内容は `~/.sdlocal/config` に保存されます。
+
+## 使い方
+
+### setサブコマンド
+key/value形式で設定を行います。ビルドを実行するには`api-url`, `store-url`, `token`を設定する必要があります。
+利用できる設定内容については、[設定できるキーの一覧](#設定できるキーの一覧)を参照ください。
```
Comment (tk3fftk) [inline docs/ja/user-guide/local.md]: 読んでいて `参照ください` はちょっと違和感がありました。
`ご参照ください`か`参照してください` な気がします。

File: docs/ja/user-guide/local.md [hunk 21]
```diff
@@ -0,0 +1,184 @@
...
+### env-fileの形式
+`--env-file` オプションで指定するファイルは docker の [.env](https://docs.docker.com/compose/env-file/) と同様の形式です。
+
+- ファイルの各行は `VAR=VAL` の形式で構成されます
+- `#` で始まる行はコメントとして無視されます
+- 空行は無視されます
+- 引用符に特殊な処理はされません。値の一部とみなされます
+
+## 環境変数の一覧
+sd-localのビルド内では、以下の環境変数が利用できます。デフォルト値に`-`が設定されているものは、Screwdriver上でのビルドと同様のため、[環境変数](environment-variables)を参照ください。
```
Comment (tk3fftk) [inline docs/ja/user-guide/local.md]: ここも

File: docs/ja/user-guide/local.md [hunk 22]
```diff
@@ -0,0 +1,184 @@
...
+
+```bash
+$ sd-local config view
+```
+
+### 設定できるキーの一覧
+
+|キー|説明|
+|---|---|
+|api-url|ご利用しているクラスタのScrewdriver APIのURLを指定してください|
```
Comment (tk3fftk) [inline docs/ja/user-guide/local.md]: `ご利用しているクラスタ`  -> `ご利用のクラスタ` or `利用中のクラスタ` が自然かもしれません

File: docs/ja/user-guide/local.md [hunk 23]
```diff
@@ -0,0 +1,184 @@
...
+
+|環境変数名|デフォルト値|
+|---|---|
+|SCREWDRIVER|false|
+|CI|false|
+|CONTINUOUS_INTEGRATION|false|
+|SD_API_URL|configで設定したapi-urlが設定されます|
+|SD_STORE_URL|configで設定したstore-urlが設定されます|
+|SD_TOKEN|configで設定したtokenに基づいたJWTが設定されます|
+|SD_SOURCE_DIR|/sd/workspace/src/screwdriver.cd/sd-local/local-build|
```
Comment (tk3fftk) [inline docs/ja/user-guide/local.md]: Should be this directory backquoted?

File: docs/ja/user-guide/local.md [hunk 24]
```diff
@@ -0,0 +1,184 @@
...
+$ chmod +x /usr/local/bin/sd-local
+```
+
+# クイックスタート
+sd-localでビルドを実行するまでの流れを説明します。
+
+ここでは、ScrewdriverのAPIを `https://api.screwdriver.cd`、 Storeを `https://store.screwdriver.cd` とします。
+
+## APIトークンの作成
+sd-localでは、Screwdriver API, Storeとやり取りするためのAPIトークンを利用します。
```
Comment (ibu1224) [inline docs/ja/user-guide/local.md]: `sd-localは、Screwdriver API, Storeとやり取りするためにAPIトークンを利用します。`
こちらの表現の方が違和感がないと思いますがどうでしょう？

File: docs/ja/user-guide/local.md (hunk 20 above)
Comment (sugarnaoming) [reply docs/ja/user-guide/local.md]: 修正しました。

File: docs/ja/user-guide/local.md (hunk 21 above)
Comment (sugarnaoming) [reply docs/ja/user-guide/local.md]: 同じく

File: docs/ja/user-guide/local.md (hunk 22 above)
Comment (sugarnaoming) [reply docs/ja/user-guide/local.md]: こちらも修正しました

File: docs/ja/user-guide/local.md (hunk 23 above)
Comment (sugarnaoming) [reply docs/ja/user-guide/local.md]: I fixed

File: docs/ja/user-guide/local.md (hunk 24 above)
Comment (sugarnaoming) [reply docs/ja/user-guide/local.md]: 修正しました

**Review by tk3fftk — APPROVED** [review]
:+1: for Japanese guide

File: docs/ja/user-guide/local.md (hunk 24 above)
Comment (ibu1224) [reply docs/ja/user-guide/local.md]: 👍

File: docs/ja/user-guide/local.md [hunk 25]
```diff
@@ -0,0 +1,184 @@
...
+|-e, --env|ビルド環境に設定する環境変数を`<key>=<value>` 形式で設定します（複数指定可）|
+|--env-file|ビルド環境に設定する環境変数をファイル形式で設定します（ファイルの形式は[env-fileのフォーマット](#env-fileの形式)を参照）|
+|-m, --memory|ビルド環境のメモリリソースのリミット値を指定します（b,k,m,gのメモリ単位で指定できます）|
+|--meta|ビルド環境に渡す[メタデータ](metadata)を指定します（JSON形式）例: `"{\"HOGE\": \"FOO\"}"`|
+|--meta-file|ビルド環境に渡す[メタデータ](metadata)をファイルで指定します（JSON形式）|
+|--src-url|ソースコードをリモートのSCMから取得します（`https` または `ssh` 形式のURLを渡せます）|
+
+注意：  
+- `--env`, `--env-file` オプションで同じ環境変数が指定された場合、 `--env` で指定した環境変数が優先されます
+- `-meta`, `--meta-file` オプションを同時に指定することはできません
```
Comment (tkyi) [inline docs/ja/user-guide/local.md]:
```suggestion
- `--meta`, `--meta-file` オプションを同時に指定することはできません
```
missing a dash?

**Review by tkyi — APPROVED** [review]
👍

**Review by jithine — APPROVED** [review]

**Pull request merged**